onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -397,17 +397,37 @@ void AttentionTypeAndShapeInference(ONNX_NAMESPACE::InferenceContext& ctx, int p
 
     auto& bias_shape = getInputShape(ctx, 2);
     auto& bias_dims = bias_shape.dim();
-    if (bias_dims.size() != 1 || bias_shape.dim(0).dim_value() % 3 != 0) {
+    if (bias_dims.size() != 1) {
       fail_shape_inference("Invalid bias shape");
     }
 
+    std::vector<int64_t> qkv_hidden_sizes;
+    getRepeatedAttribute(ctx, "qkv_hidden_sizes", qkv_hidden_sizes);
+
+    if (hasInputShape(ctx, extra_add_index)) {
+      auto& extra_add_shape = getInputShape(ctx, extra_add_index);
+      auto& extra_add_dims = extra_add_shape.dim();
+      if (extra_add_dims.size() != 4) {
+        fail_shape_inference("Extra add input should have 4 dimenstions");
+      }
+    }
+
+    int64_t output_hidden_size;
+    if (qkv_hidden_sizes.size() != 0) {
```

Comment:
Better to change to qkv_hidden_sizes.size() == 3. Otherwise, there could be exception in qkv_hidden_sizes[2].

Reply:
updated to check the size and get the output_hidden_size.